src/Microsoft.Health.Dicom.Api/Logging/TelemetryInitializer.cs

```diff
@@ -18,16 +24,29 @@ namespace Microsoft.Health.Dicom.Api.Logging;
 internal class TelemetryInitializer : ITelemetryInitializer
 {
     private readonly IHttpContextAccessor _httpContextAccessor;
+    private readonly bool _enableDataPartitions;
+    private readonly bool _enableExport;
+    private readonly bool _enableExternalStore;
     private const string ApiVersionColumnName = "ApiVersion";
+    private const string EnableDataPartitions = "EnableDataPartitions";
+    private const string EnableExport = "EnableExport";
+    private const string EnableExternalStore = "EnableExternalStore";
+    private const string UserAgent = "UserAgent";
 
-    public TelemetryInitializer(IHttpContextAccessor httpContextAccessor)
+    public TelemetryInitializer(IHttpContextAccessor httpContextAccessor, IOptions<FeatureConfiguration> featureConfiguration)
     {
         _httpContextAccessor = EnsureArg.IsNotNull(httpContextAccessor, nameof(httpContextAccessor));
+        EnsureArg.IsNotNull(featureConfiguration?.Value, nameof(featureConfiguration));
+        _enableDataPartitions = featureConfiguration.Value.EnableDataPartitions;
+        _enableExport = featureConfiguration.Value.EnableExport;
+        _enableExternalStore = featureConfiguration.Value.EnableExternalStore;
     }
 
     public void Initialize(ITelemetry telemetry)
     {
         AddApiVersionColumn(telemetry);
+        if (telemetry is RequestTelemetry requestTelemetry)
+            AddPropertiesFromHttpContextItems(requestTelemetry);
     }
 
     private void AddApiVersionColumn(ITelemetry telemetry)
```

Comment:
Should we rename this method now since it is not just populating ApiVersionColumn now?